### PR TITLE
standard repo files:

### LINKED ISSUE #1: Standard Stuff
Make sure the plugin has all the standard stuff like:
- [ ] readme
- [ ] changelog
- [ ] rake
- [ ] travis
- [ ] rubygems

### PR DESCRIPTION
**Pull request by majormoses**
- `.travis.yml`
- `.rubocop.yml`
- `Rakefile`
- `CHANGELOG.md`

## Pull Request Checklist

closes #1

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [ ] Existing tests pass 

#### Purpose
Setup the standard stuff for deployment
#### Known Compatibility Issues
None

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,42 +1,31 @@
...
     on_failure: always
 script:
-  - rvm use --default $TRAVIS_RUB_VERSION
-  - echo 'export PATH="$PATH:/home/travis/.rvm/bin"' | sudo tee -a /etc/profile
-  - echo 'source /home/travis/.rvm/scripts/rvm' | sudo tee -a /etc/profile
-  - bundle exec rake default
-  - bundle exec kitchen test
-  - sudo chown -R travis:travis /home/travis/.rvm
-  - gem build sensu-plugins-dcos.gemspec
-  - gem install sensu-plugins-dcos-*.gem
```
Comment (eheydrick) [inline .travis.yml]: the kitchen integration test looks useful - i think we should leave that.

Comment (majormoses) [inline .travis.yml]: good call, thats what I get for just copying the files over from skel.

Comment (luisdavim) [inline .travis.yml]: yes, let's keep the kitchen test part.

File: bin/check-dcos-ping.rb [hunk 2]
```diff
@@ -57,11 +57,10 @@ class CheckDcosPing < Sensu::Plugin::Check::CLI
 
   def run
     value = get_data(config[:url])['ok']
-    message "OK = #{value}"
```
Comment (luisdavim) [inline bin/check-dcos-ping.rb]: why not use `message`? the way it is at the moment if the health check fails the message is `CheckDcosPing CRITICAL: OK = false` if it passes it's `CheckDcosPing OK: OK = true`, this is the way DC/OS reports the issue, it returns a json string like this `{"OK": "true"}`

**Comment by luisdavim** [comment]
and I agree with @eheydrick that we should delete the bitbucket-pipelines.yml and s3_upload.rb files, those are leftovers that I forgot to cleanup.

File: bin/check-dcos-ping.rb (hunk 2 above)
Comment (majormoses) [inline bin/check-dcos-ping.rb]: If you prefer I can put that part back, what I don't like and prompted this was the case inequality which needlessly complicates code.

**Comment by luisdavim** [comment]
looks like travis doesn't like kitchen:
```
The command "bundle exec rake default" exited with 0.
0.56s$ bundle exec kitchen test
-----> Starting Kitchen (v1.16.0)
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::UserError
>>>>>> Message: Vagrant 1.1.0 or higher is not installed. Please download a package from http://www.vagrantup.com/downloads.html.
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by majormoses** [comment]
we could try using kitchen dokken to avoid the dependency on vagrant.

**Comment by majormoses** [comment]
ill spend some time tonight to play with this.

**Comment by luisdavim** [comment]
Hi @majormoses, I've never used kitchen-docker with travis myself bu I found this https://github.com/zuazo/kitchen-in-travis that might help.

**Comment by majormoses** [comment]
@luisdavim ill take a look, the problem I seem to be having is that the volume I want mounted (travis rubies) is not mounting for some reason and I haven't really had the time to sit down and get this done. I have used both kitchen-docker and kitchen-dokken with travis before but this is a very different setup.

**Comment by majormoses** [comment]
Well apparently that's a feature that travis does not support at the moment: https://github.com/travis-ci/worker/issues/171

File: test/fixtures/bootstrap.sh [hunk 3]
```diff
@@ -118,5 +129,5 @@ echo "
 service nginx restart
 
 cd $DATA_DIR
-SIGN_GEM=false $RUBY_HOME/bin/gem build sensu-plugins-dcos.gemspec
-$RUBY_HOME/bin/gem install sensu-plugins-dcos-*.gem
+SIGN_GEM=false gem build sensu-plugins-dcos.gemspec
```
Comment (eheydrick) [inline test/fixtures/bootstrap.sh]: no need for `SIGN_GEM=false`. that is leftovers from back when the gems were signed.

File: .kitchen.yml [hunk 4]
```diff
@@ -1,17 +1,31 @@
...
+  #     image: ruby:2.2-slim
+  # - name: ruby-230
+  #   driver:
+  #     image: ruby:2.3.0-slim
+  # - name: ruby-220
+  #   driver:
+  #     image: ruby:2.2-slim
   - name: default
+    driver:
+      image: ruby:2.4.1-slim
```
Comment (eheydrick) [inline .kitchen.yml]: if would be nice if there was a way to to drive the ruby version based on the ruby version being tested in travis.

**Comment by eheydrick** [comment]
Overall looks good, left a few comments.

File: .kitchen.yml (hunk 4 above)
Comment (majormoses) [inline .kitchen.yml]: agreed, I was originally hoping to use just a default suite and symlink them but that apparently does not work nor does mounting a docker volume. I am planning on basically converting everything to shared examples and then in each suite we basically just call the shared examples and thats it.

**Comment by majormoses** [comment]
@eheydrick @luisdavim I am gonna merge this and do some enhancement later for the multiple suites and more shared specs.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-dcos/versions/0.0.3

**Pull request merged**